intruder_alarm/src/singly/mod.rs

```diff
@@ -0,0 +1,220 @@
+#[derive(Default)]
+pub struct List<T, N, R> {
+    /// Link to the head node of the list.
+    head: Link<N>,
+
+    /// Length of the list.
+    len: usize,
+
+    /// Type marker for items stored in the list.
+    _elem_ty: PhantomData<T>,
+
+    /// Type marker for the `OwningRef` type.
+    _ref_ty: PhantomData<R>,
+}
+
+//  Linked
+/// Trait that must be implemented in order to be a member of an intrusive
+/// singly linked list.
+pub trait Linked: Sized {
+    /// Borrow this element's next [`Link`].
+    ///
+    /// [`Links`]: struct.Link.html
+    fn next(&self) -> &Link<Self>;
+
+    /// Mutably borrow this element's next [`Link`].
+    ///
+    /// [`Links`]: struct.Link.html
+    fn next_mut(&mut self) -> &mut Link<Self>;
+
+    /// De-link this node, returning its' next Link.
+    fn take_next(&mut self) -> Link<Self> {
+        mem::replace(self.next_mut(), Link::none())
+    }
+}
+
+//-----------------------------------------------------------------------------
+// Implementations
+//-----------------------------------------------------------------------------
+
+// ===== impl List =====
+
+impl<T, Node, R> List<T, Node, R> {
+    /// Create a new `List` with 0 elements.
+    pub const fn new() -> Self {
+        List {
+            head: Link::none(),
+            len: 0,
+            _elem_ty: PhantomData,
+            _ref_ty: PhantomData,
+        }
+    }
+
+    /// Returns the length of the list.
+    #[inline]
+    pub fn len(&self) -> usize {
+        self.len
+    }
+
+    /// Returns true if the list is empty, false otherwise.
+    #[inline]
+    pub fn is_empty(&self) -> bool {
+        self.len == 0
+    }
+
+    /// Borrows the first node of the list as an `Option`.
+    ///
+    /// Note that this is distinct from `front`: this method
+    /// borrows the head _node_, not the head _element_.
+    ///
+    /// # Returns
+    ///   - `Some(&N)` if the list has elements
+    ///   - `None` if the list is empty.
+    #[inline]
+    pub fn head(&self) -> Option<&Node> {
+        self.head.as_ref()
+    }
+
+    /// Mutably borrows the head node of the list as an `Option`
+    ///
+    /// # Returns
+    ///   - `Some(&mut Node)` if the list has elements
+    ///   - `None` if the list is empty.
+    #[inline]
+    pub fn head_mut(&mut self) -> Option<&mut Node> {
+        self.head.as_mut()
+    }
+}
+
+impl<T, Node, Ref> List<T, Node, Ref>
+where
+    Node: Linked,
+    Ref: OwningRef<Node>,
+    Ref: DerefMut,
+{
+    /// Push a node to the list.
+    pub fn push(&mut self, mut node: Ref) -> &mut Self {
+        unsafe {
+            *node.next_mut() = self.head;
+            let node = Link::from_owning_ref(node);
+            self.head = node;
+            self.len += 1;
+        };
+        self
+    }
+}
+
+impl<T, Node, Ref> List<T, Node, Ref>
+where
+    Node: Linked,
+    Ref: OwningRef<Node>,
+{
+    /// Pop a node from the list.
+    pub fn pop(&mut self) -> Option<Ref> {
```

Comment:
For consistency with the doubly-linked list API, I would probably call this `pop_node`.